### PR TITLE
Fix issues in accurateCastOrNull

### LINKED ISSUE #38585: accurateCastOrNull raises an exception, rather than simply returning null when casting an invalid value to Bool
**Describe what's wrong**

accurateCastOrNull raises an exception, rather than simply returning null when casting an invalid value to Bool

```
Code: 467. DB::Exception: Received from 127.0.0.1:9000. DB::Exception: Cannot parse boolean value here: 'test', should be 'true' or 'false' controlled by setting bool_true_representation and bool_false_representation or one of True/False/T/F/Y/N/Yes/No/On/Off/Enable/Disable/Enabled/Disabled/1/0: While processing accurateCastOrNull('test', 'Bool'). (CANNOT_PARSE_BOOL)
```

**Does it reproduce on recent release?**

ClickHouse server version 22.7.1.906 (official build).

**How to reproduce**

```
select accurateCastOrNull('test', 'Bool')
```

**Expected behavior**

```
I expect it to behave the same as an invalid cast to something like UInt8

SELECT accurateCastOrNull('test', 'UInt8')

Query id: e7d0473b-6ae0-455d-9eae-fb3827e7c671

┌─accurateCastOrNull('test', 'UInt8')─┐
│                                ᴺᵁᴸᴸ │
└─────────────────────────────────────┘
```

### PR DESCRIPTION
**Pull request by aiven-sal**
`accurateCastOrNull` must not throw when casting an arbitrary string to Bool.

Close #38585

This PR also fixes an unrelated issue that caused `accurateCastOrNull` to behave like `accurateCastOrDefault` on error when casting to IPv4 or IPv6.   

### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
* accurateCastOrNull will not throw when casting an arbitrary string to Bool and it'll return Null
* accurateCastOrNull will return Null instead of the all zeroes addresses, if an invalid string is casted to an IP
* When cast_keep_nullable=1 casting from a nullable empty string to IP will produce Null as it happens with all of the other types

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

File: src/Functions/FunctionsConversion.h [hunk 1]
```diff
@@ -1597,7 +1597,20 @@ struct ConvertImplGenericFromString
 
                 const auto & val = col_from_string->getDataAt(i);
                 ReadBufferFromMemory read_buffer(val.data, val.size);
-                serialization_from.deserializeWholeText(column_to, read_buffer, format_settings);
+                try
+                {
+                    serialization_from.deserializeWholeText(column_to, read_buffer, format_settings);
+                }
+                catch (Exception &)
```
Comment (alexey-milovidov) [inline src/Functions/FunctionsConversion.h:1604]: This is wrong because there can be unrelated exceptions, such as memory errors.

Comment (alexey-milovidov) [inline src/Functions/FunctionsConversion.h:1604]: Missing const.

**Comment by robot-ch-test-poll1** [comment]
This is an automated comment for commit bf47ac5f535e058eba7401d6379be1fdef201732 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/54136/bf47ac5f535e058eba7401d6379be1fdef201732/ci_running.html)
The overall status of the commit is 🔴 failure

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🔴 failure</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

**Review by alexey-milovidov — APPROVED** [review]
Yes, rethrowing non-specific exceptions is enough.

**Comment by alexey-milovidov** [comment]
@aiven-sal Fuzzer has found an error, PTAL.

**Pull request merged**